### ISSUE TITLE
Incorrect groupby displayed in IPython Notebook with 64-bit floats and multi-index

### ISSUE THREAD

**Issue by brianckeegan** [pydata/pandas#10796]
I ran into this bug when attempting some groupby-aggregates on data that had 64-bit floats.

    _ids = np.array(range(201500000100,201500000110)*3).astype(float)
    _df = pd.DataFrame({'id':_ids,
              'value_1':np.random.randint(0,10,30),
              'value_2':list('ABC')*15})

A simple groupby-aggregation produces the correct results, namely returning a DataFrame with an index of 10 rows.

    _gb1 = _df.groupby('id').agg({'value_1':np.sum})

value_1 | id
----------- | -----------
2.015000e+11	| 14
2.015000e+11	| 4
2.015000e+11	| 8
2.015000e+11	| 25
2.015000e+11	| 12
2.015000e+11	| 16
2.015000e+11	| 15
2.015000e+11	| 21
2.015000e+11	| 8
2.015000e+11	| 7

However, grouping by the second value to return a multi-index returns an unexpected DataFrame. 

    _df.groupby(['id','value_2']).agg({'value_1':np.sum})

id | value_2 | value_1 
----------- | ----------- | -----------
2.015000e+11|A|2
 |B|5
 |C|3
 |A|3
 |B|0
 |C|0
 |A|4
 |B|1
 |C|3
 |A|8
 |B|7
 |C|0
 |A|3
 |B|0
 |C|2
 |A|0
 |B|4
 |C|5
 |A|4
 |B|9
 |C|0
 |A|6
 |B|6
 |C|6
 |A|8
 |B|3
 |C|4
 |A|5
 |B|7
 |C|3

Casting the `id` variable to unicode appears to correct the problem and return an aggregated DF with the correct dimensions and multi-indexing.
    
    _df['id_str'] = _df['id'].astype(unicode).str.slice(0,-2)
    _df.groupby(['id_str','value_2']).agg({'value_1':np.sum})

----

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.10.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 69 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.16.2
nose: 1.3.7
Cython: 0.22.1
numpy: 1.9.2
scipy: 0.16.0
statsmodels: 0.6.1
IPython: 3.2.1
sphinx: 1.3.1
patsy: 0.3.0
dateutil: 2.4.2
pytz: 2015.4
bottleneck: 1.0.0
tables: 3.2.0
numexpr: 2.4.3
matplotlib: 1.4.3
openpyxl: 1.8.5
xlrd: 0.9.3
xlwt: 1.0.0
xlsxwriter: 0.7.3
lxml: 3.4.4
bs4: 4.3.2
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.5
pymysql: None
psycopg2: None

**Comment by jreback** [comment]
this is really odd to use a datelooking value as a float. why are you doing this? or are these actual values?

**Comment by brianckeegan** [comment]
They're ID values bucketed by year, but with enough head room to allow millions of users in the future. So they're not datetimes and they're stored in a server as floats.

**Comment by jreback** [comment]
ok, this is a bug in the MultiIndex index display as the index itself looks ok (and the first example yields a ``FloatIndex`` which looks fine).

pull-requests are welcome

**Comment by TomAugspurger** [comment]
This looks like the same issue, right?

```python
In [2]: df = pd.DataFrame({"A": [1, '1'], 'B': [2, 2], 'C': [3, 4]})

In [3]: df
Out[3]:
   A  B  C
0  1  2  3
1  1  2  4

In [11]: df.set_index(['A', 'B'])
Out[11]:
     C
A B
1 2  3
  2  4
```

Maybe a bit easier to see. The outer level should have `[1, '1']` (added an apostrophe to be explicit).

**Comment by jreback** [comment]
yeah, that's a formatting issue in the sparsifier. Further *maybe* we should actually put quotes around strings if we have a mixed column

```
In [14]: pd.lib.infer_dtype(df.A)
Out[14]: 'mixed-integer'
```
maybe display

```
In [11]: df
Out[11]: 
     A  B  C
0    1  2  3
1  '1'  2  4
```
